### ISSUE TITLE
[Bug] SageMaker fails to start due to policy issue.

### ISSUE THREAD

**Issue by BoPeng** [awslabs/service-workbench-on-aws#1176]
**Describe the bug**
When I start a sagemaker workspace, it fails with an error message. As far as I remember, sagemaker was able to start a few days ago.

**To Reproduce**
1. regular SWB installation
2. start sagemaker,

**Expected behavior**
SageMaker should start

**Screenshots**
![image](https://user-images.githubusercontent.com/9889312/232597018-ada430b8-0928-4ca8-90fc-9495295b0632.png)

The relevant error message from CloudFormation is

```
User: arn:aws:sts::948662388200:assumed-role/prod-oh-sw-LaunchConstraint/servicecatalog is not authorized to perform: sagemaker:CreateNotebookInstanceLifecycleConfig on resource: arn:aws:sagemaker:us-east-2:948662388200:notebook-instance-lifecycle-config/BasicNotebookInstanceLifecycleConfig-pR9dIu4ZjVhk because no identity-based policy allows the sagemaker:CreateNotebookInstanceLifecycleConfig action (Service: AmazonSageMaker; Status Code: 400; Error Code: AccessDeniedException; Request ID: ca7030e3-8810-4b6e-9e90-fe062eed3bca; Proxy: null)
```

**Versions (please complete the following information):**
 - Release Version installed: latest master
 - Is the deployment from a forked version of the repository? no.

**Additional context**
1. I checked service catelog, the product is associated with a role: `prod-oh-sw-LaunchConstraint`
2. The IAM role `prod-oh-sw-LaunchConstraint` exists and with policy `SageMakerLaunchPermissions`, which contains lines
```yaml
        {
            "Action": [
                "sagemaker:DescribeNotebookInstanceLifecycleConfig",
                "sagemaker:CreateNotebookInstanceLifecycleConfig",
                "sagemaker:DeleteNotebookInstanceLifecycleConfig"
            ],
            "Resource": "arn:aws:sagemaker:*:*:notebook-instance-lifecycle-config/basicnotebookinstancelifecycleconfig-*",
            "Effect": "Allow"
        },
```

So the policy does exist. I do not know what else have gone wrong. 
3. The only thing I did in between was perhaps adding a role to my account that allows me to assume admin role of another account, which should not affect this existing role.

What else should I do to debug?

**Comment by kpark277** [comment]
Hi BoPeng, SageMaker has released an update that makes the policy case sensitive. The current workaround is to update the policy to use BasicNotebookInstanceLifecycleConfig-* and BasicNotebookInstance-* for the resources in that policy. We are currently working on a PR that will fix this once the HostingAccount and MainAccount templates are updated through CloudFormation.

**Comment by BoPeng** [comment]
Thanks! I was suspecting something like that but I do not know AWS well enough to know if policies are case sensitive. I will manually change the policies.

**Comment by kpark277** [comment]
Here is the PR that should fix the issue: https://github.com/awslabs/service-workbench-on-aws/pull/1177
Note that we opted for ? instead of the exact case to make the change backwards compatible with both capitalized and uncapitalized letters.

**Comment by BoPeng** [comment]
I can confirm that the fix in #1177 works.
